File: modified_scripts/r/bubble_sort/gemma_7b.r

bubble_sort <- function(arr) {
  for (i in 1:length(arr) - 1) {
    for (j in i + 1:length(arr)) {
      if (arr[i] > arr[j]) {
        arr[i] <- arr[j]
        arr[j] <- arr[i]
      }
    }
  }
  arr
}

File: modified_scripts/rust/bubble_sort/gemma_7b.rs

fn bubble_sort(arr: &mut [i32]) {
    for i in 0..(arr.len() - 1) {
        for j in 0..(arr.len() - i - 1) {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
            }
        }
    }
}
